### PR TITLE
#880 Error message when uploading files with short names in not meaningful 

### PR DESCRIPTION
**Pull request by kujtimiihoxha**
For #880
 -Added file length check to show specific message to user.

### CODE REVIEW

**Comment by karato** [comment]
@kujtimiihoxha Thanks, someone will review your pull request soon

**Comment by karato** [comment]
@krzyk do this review

File: netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java [hunk 1]
```diff
@@ -61,7 +61,14 @@
      * Table name.
      */
     public static final String TBL = "attachments";
-
+    /**
+     * Min number of char.
+     */
+    public static final int MIN = 3;
+    /**
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java]: @kujtimiihoxha please add new lines between fields

File: netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java [hunk 2]
```diff
@@ -197,7 +204,17 @@ public int unseen() throws IOException {
...
+                || name.length() > DyAttachments.MAX) {
+            throw new Attachments.InvalidNameException(
+                String.format(
+                        "invalid attachment name length \"%s\". %s%s",
+                        name,
+                        "The length of the attachment name must be ",
+                        "between 3-100 characters"
+                )
+            );
+        } else if (!name.matches("[a-zA-Z\\.\\-0-9]{3,100}")) {
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java]: @kujtimiihoxha please read http://www.yegor256.com/2015/01/21/if-then-throw-else.html

File: netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java [hunk 3]
```diff
@@ -197,7 +204,17 @@ public int unseen() throws IOException {
 
     @Override
     public void create(final String name) throws IOException {
-        if (!name.matches("[a-zA-Z\\.\\-0-9]{3,100}")) {
+        if (name.length() < DyAttachments.MIN
+                || name.length() > DyAttachments.MAX) {
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java]: @kujtimiihoxha indentation should be 4 spaces less

File: netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java [hunk 4]
```diff
@@ -197,7 +204,17 @@ public int unseen() throws IOException {
...
+            throw new Attachments.InvalidNameException(
+                String.format(
+                        "invalid attachment name length \"%s\". %s%s",
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/dynamo/DyAttachments.java]: @kujtimiihoxha indentation should be 4 spaces less here and in the lines below

**Comment by krzyk** [comment]
@kujtimiihoxha please see my comments, and also you forgot to add a test

**Comment by kujtimiihoxha** [comment]
@krzyk  I hope my solution satisfies  http://www.yegor256.com/2015/01/21/if-then-throw-else.html, I will add the test case as soon as possible. About the layout.xls I accidentally committed this file it has to do with another issue I solved.I updated it now from master so it will not change anything in the master branch

**Comment by krzyk** [comment]
@kujtimiihoxha looks good, but the layout.xsl still contains a change, please revert it.

**Comment by kujtimiihoxha** [comment]
@krzyk Is everything OK now ?

File: netbout-web/src/test/java/com/netbout/dynamo/DyAttachmentsITCase.java [hunk 5]
```diff
@@ -74,4 +74,35 @@ public void createsAndLoadsAttachments() throws Exception {
         attachments.delete(name);
     }
 
+    /**
+     * DyAttachments can create, save and load attachments.
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/dynamo/DyAttachmentsITCase.java]: @kujtimiihoxha wrong description

File: netbout-web/src/test/java/com/netbout/dynamo/DyAttachmentsITCase.java [hunk 6]
```diff
@@ -74,4 +74,35 @@ public void createsAndLoadsAttachments() throws Exception {
...
+        final Aliases aliases =
+            new DyBase().user(new URN("urn:test:840919")).aliases();
+        aliases.add(alias);
+        final Inbox inbox = aliases.iterate().iterator().next().inbox();
+        final Bout bout = inbox.bout(inbox.start());
+        final Attachments attachments = bout.attachments();
+        final String name = "a";
+        try {
+            attachments.create(name);
+        } catch (final Attachments.InvalidNameException ex) {
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/dynamo/DyAttachmentsITCase.java]: @kujtimiihoxha you should also make sure that the exception will be thrown, e.g. by adding an `Assert.fail(...)` in the try.

**Comment by krzyk** [comment]
@kujtimiihoxha one more comment

**Comment by krzyk** [comment]
@kujtimiihoxha ping

**Comment by kujtimiihoxha** [comment]
@krzyk updated

**Comment by krzyk** [comment]
@rultor merge pls

File: netbout-web/src/main/xsl/layout.xsl [hunk 7]
```diff
@@ -122,8 +122,9 @@
                         </li>
                         <xsl:if test="links/link[@rel='start']">
                             <li>
+
                                 <xsl:choose>
-                                    <xsl:when test="/page/bouts and count(/page/bouts/bout) = 0 and /page/query = ''">
+                                    <xsl:when test="/page/bouts and count(/page/bouts/bout) = 0 and /page/query = '' and not(contains(version/name,'LOCAL'))">
```
Comment (dmzaytsev) [inline netbout-web/src/main/xsl/layout.xsl]: @kujtimiihoxha I didn't get why you change this line?

**Comment by dmzaytsev** [comment]
@kujtimiihoxha please see one question above

**Comment by kujtimiihoxha** [comment]
@dmzaytsev This file was commiged by accident i made it the same as master branch.

**Comment by dmzaytsev** [comment]
@kujtimiihoxha did you merge with master?

**Comment by dmzaytsev** [comment]
@kujtimiihoxha I guess Rultor will reject this merge request
please merge with master before

**Comment by kujtimiihoxha** [comment]
@dmzaytsev  I have merged with master now.

**Comment by dmzaytsev** [comment]
@kujtimiihoxha thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@elenavolokhova please, review this task for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor deploy pls

**Comment by elenavolokhova** [comment]
@karato Looks good!

**Comment by karato** [comment]
> @karato Looks good!

@elenavolokhova thanks for the review

**Comment by karato** [comment]
@krzyk I added 10 mins to @elenavolokhova (for QA review) in transaction 75907777; quality is good here, 10 mins added to @dmzaytsev (our architect) in transaction 75907797; thank you, added **26 mins** to your acc, payment num is `75907826`, 147 hours and 29 mins spent in total to complete; you're getting extra minutes here (c=11); +26 added to your rating, at the moment it is: [+4574](http://www.netbout.com/b/35035?open=rating)

**Pull request merged**